spec/controller/product_spec.rb

```diff
@@ -18,4 +23,79 @@
       expect(response).to render_template(:show)
     end
   end
+
+  describe 'GET #new' do
+    it 'renders the new template' do
+      get :new
+      expect(response).to render_template(:new)
+    end
+  end
+
+  describe 'POST #create' do
+    context 'with valid attributes' do
+      it 'creates a new product' do
+        expect {
+          post :create, params: { product: FactoryBot.attributes_for(:product, category_id: category.id) }
+        }.to change(Product, :count).by(1)
+      end
```

Comment:
Test the created object as well. 
Like 
Expected Product name
Expected Product Category